src/irmin/object_graph.ml

```diff
@@ -207,11 +208,11 @@ module Make (Hash : HASH) (Branch : Type.S) = struct
         keys
     in
     let visit key level =
-      if level >= depth then Lwt.return_unit
+      if (* ISSUE *) level >= depth then Lwt.return_unit
```

Comment:
`depth = 0` currently implies we don't visit anything, it feels counter-intuitive to me. Could we test `level > depth` instead and rename `depth` to `max_depth`?